afew/tests/test_mailmover.py

```diff
@@ -88,10 +89,12 @@ def tearDown(self):
 
     @staticmethod
     def get_folder_content(db, folder):
-        return {
-            (os.path.basename(msg.get_message_id()), msg.get_part(1).decode())
-            for msg in db.do_query('folder:{}'.format(folder)).search_messages()
-        }
+        ret = set()
+        for msg in db.open().messages('folder:{}'.format(folder)):
+            with open(msg.path) as f:
+                ret.add((os.path.basename(msg.messageid),
+                         email.message_from_file(f).get_payload()))
+        return ret
```

Comment:
Question: would it be interesting to run this method asynchronously to speed things up? I believe this will speed things up when someone has a bunch of emails to process. Perhaps having a private method that handles the asynchronous call and calls `add`? I've done something similar here: https://github.com/afewmail/afew/pull/316/files#r744197423

Reply:
This is just in the test suite, which has 10 e-mails or so.

That mail mover test suite is not exactly "pretty" anyways, the proposed changes are the minimal ones to let it pass with the notmuch2 bindings. I would refactor the e-mail comparisons quite a bit, for example.